modules/codegen/src/main/scala/com/twilio/guardrail/SwaggerUtil.scala

```diff
@@ -251,9 +251,9 @@ object SwaggerUtil {
             case ("boolean", fmt)              => booleanType(fmt).map(log(fmt, _))
             case ("array", fmt)                => arrayType(fmt).map(log(fmt, _))
             case ("file", fmt) =>
-              fileType(fmt).map(log(fmt, _))
-            case ("binary", _) =>
-              fileType(None).map(log(None, _))
+              fileType(None).map(log(fmt, _))
```

Comment:
Why is this `None` required here? Logging `fmt` but passing `None` is misleading

Reply:
binary has default format `binary` and then is rendered as type `binary`, which doesn't compile ... would you prefer to change signature if filetype to not take any format? I'm not sure what kind of format files can have anyway and i didn't manage to find any format definition

Reply:
well, this is unfortunate.

It turns out [`type: string`, `format: binary`](https://swagger.io/specification/#dataTypes) (as well as `type: string`, `format: byte`). I hope that the 2.x -> 3.x handles this mapping for us 🤔

Reply:
yep, do you want me to add support for type: byte also?

Reply:
I think that's very much scope creep, attempting to deal with the Base64 encoding issue 😁

That being said, I'd appreciate verifying that both swagger 2.x and 3.x behave the same way here. As `binary` has been demoted to a `format` of `string`, this should definitely not be `case ("binary", None)`, it should be `case ("string", "binary")` and we should ensure that both 2.x and 3.x generate the same code (as I mentioned earlier).

Reply:
well i was not gonna solve that issue, i was just gonna add code that will render `format` `byte` as file so that users are not completely screwed and can use it on their own

i don't understand what you mean by `behave the same way here` - do you mean that swagger 2.0 defintion with `type:file` generates the same code as 3.0 definition with `type: string format: binary`?

Reply:
Ah -- rendering `type: string; format: byte` is almost certainly to ensure that non-latin1 charactersets are not destroyed in transit. This is apparently a common technique in some non-english speaking countries. Treating them as `File` would definitely be inconvenient.

"behave the same way" meaning ensuring that they generate the same code, that is correct. I think having a regression test for the same `definition` in both OpenAPI 2.x and OpenAPI 3.x then comparing the `structure`s they produce would be overkill, considering that this is behaviour that should be handled in `swagger-parser-v2-converter`. Just verifying that it works at all would be sufficient for moving forward here (changing the `type: binary` to `type: string`, `format: binary`)

Reply:
Whoops, https://github.com/swagger-api/swagger-parser/blob/master/modules/swagger-parser-v2-converter/src/main/java/io/swagger/v3/parser/converter/SwaggerConverter.java only has one place that has the literal string `"binary"`, but that's when dealing with `type: file`.

It looks like we'll have to duplicate these patterns here.

Reply:
Please verify that this assertion is correct before working around it; I can open an issue (and possibly fix it) upstream.

Reply:
Sorry, perhaps im a little dense, but ... i assume you are talking about swagger converter because guardrail uses it internally when reading swagger 2 files ... those get converterd to openapi 3 and then guardrail works with them, is that correct? Are you sure this is happening? Because when working with `type: file`, it was represented in as `FileSchema`, not as `string with binary format` which is what i would expect if the assumption that v2 sources are converted to v3 sources

Reply:
hmm, it looks like swagger converts `type: file` to `binary string` if and only if it's as part of response: (https://github.com/swagger-api/swagger-parser/blob/master/modules/swagger-parser-v2-converter/src/main/java/io/swagger/v3/parser/converter/SwaggerConverter.java#L816)

Reply:
No need to apologise! You are correct. Internally, we use the `swagger-parser-v2-converter` in order to support both 2.x and 3.x.

Unfortunately, as that converter is bespoke and not generated via some specification of how to map 2.x to 3.x, my only reason to assume anything works is to [examine the source code in that project](https://github.com/swagger-api/swagger-parser/blob/master/modules/swagger-parser-v2-converter/src/main/java/io/swagger/v3/parser/converter/SwaggerConverter.java) to see if the functionality is there.

As for the file handling stuff you mentioned, it does appear that they explicitly handle the file type conversion. As I'm reading this converter source, it doesn't seem to do the same conversion for `"binary"`, which is why I'm asking that we verify before inadvertently removing functionality from the 2.x codebase :smile:

I would do it myself, but I'm unfortunately away from my computer until tomorrow

Reply:
yep, when reverting changes in this PR, putting type file into response fails with 
```
[info]   class FileSchema {
[info]       class Schema {
[info]           type: string
[info]           format: binary
[info]       }
[info]   }
```

while putting them into definitions it fails with:

```
[info]   class FileSchema {
[info]       class Schema {
[info]           type: file
[info]           format: binary
[info]       }
[info]   }
```

Reply:
so perheps we could split the concerns ... my main goals with this PR was to solve our issue with `type: file` in responses. I believe this pr solves those issues.

There is however unresolved issue with `type: string format:byte`, i would suggest fixing that one as different issue/pr

Reply:
I agree, `format: byte` is a different concern. Thanks for looking into this!

Reply:
sweet ... i will open issue for binary improvements tomorrow ... anything else blocking this from being merged? Thanks for quick responses btw!

Reply:
I don't think anything else is blocking this from being merged, provided that both OpenAPI 2.x and 3.x behave the same way for _all_ changed cases in `typeName`.

Reply:
well i think the `case "binary"` branch is dead and never gets called anyway (`type: binary` is not a valid in v2 or v3) and the `file` branch is only called from openapi v2 i think so it should be fine

Reply:
I think all of this may have been from a misunderstanding when I was reading the specification. I misread that `type: binary` was in the Swagger 2.0 specification.

Sorry for so much confusion! I appreciate you spelling it out to me 😁

Reply:
no worries :) if i might be so bold, could you approve the PR then? ;)